hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenClaimAirdropHandler.java

```diff
@@ -42,10 +53,46 @@ public TokenClaimAirdropHandler() {
     }
 
     @Override
-    public void preHandle(@NonNull PreHandleContext context) throws PreCheckException {}
+    public void preHandle(@NonNull PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+        final var op = context.body().tokenClaimAirdrop();
+        final var pendingAirdrops = op.pendingAirdrops();
+
+        if (pendingAirdrops.isEmpty()) {
+            throw new PreCheckException(EMPTY_PENDING_AIRDROP_ID_LIST);
+        }
```

Comment:
Because we want only computing signing requirements in the `preHadle()`(per https://github.com/hashgraph/hedera-services/pull/14279#discussion_r1685837369) and we have this check in `pureCheck()`, should we remove this check from here?